inlineplz/linters/__init__.py

```diff
@@ -55,27 +59,703 @@ def vendored_path(path):
+                ["cabal", "install", "shellcheck"],
+                ["apt-get", "install", "shellcheck"],
+                ["dnf", "install", "shellcheck"],
+                ["brew", "install", "shellcheck"],
+                ["port", "install", "shellcheck"],
+                ["zypper", "in", "ShellCheck"],
+            ],
+            "help": ["shellcheck", "-V"],
+            "run": ["shellcheck", "-f", "json"],
+            "rundefault": ["shellcheck", "-f", "json"],
+            "dotfiles": [],
+            "parser": parsers.ShellcheckParser,
+            "language": "shell",
+            "autorun": True,
+            "run_per_file": True,
+        },
+    )
+    register_linter(
+        "spotbugs-maven-plugin",
+        {
+            "install": [
+                ["mvn", "clean", "install", "-U"],
+                [
+                    "mvn",
+                    "dependency:get",
+                    "-Dartifact=com.github.spotbugs:spotbugs-maven-plugin:3.1.3",
+                ],
+            ],
+            "help": ["mvn", "com.github.spotbugs:spotbugs-maven-plugin:3.1.3:help"],
+            "run": [
+                "mvn",
+                "-Dspotbugs.failOnError=false",
+                "com.github.spotbugs:spotbugs-maven-plugin:3.1.3:check",
+            ],
+            "rundefault": [
+                "mvn",
+                "-Dspotbugs.failOnError=false",
+                "com.github.spotbugs:spotbugs-maven-plugin:3.1.3:check",
+            ],
+            "dotfiles": [],
+            "parser": parsers.SpotbugsMavenParser,
+            "language": "java",
+            "autorun": True,
+            "run_per_file": False,
+        },
+    )
+    register_linter(
+        "stylint",
+        {
+            "install": [["npm", "install", "stylint"]],
+            "help": [os.path.normpath("./node_modules/.bin/stylint"), "-h"],
+            "run": [os.path.normpath("./node_modules/.bin/stylint")],
+            "rundefault": [
+                os.path.normpath("./node_modules/.bin/stylint"),
+                "-c",
+                "{config_dir}/.stylintrc",
+            ],
+            "dotfiles": [".stylintrc"],
+            "parser": parsers.StylintParser,
+            "language": "stylus",
+            "autorun": True,
+            "run_per_file": False,
+        },
+    )
+    register_linter(
+        "yamllint",
+        {
+            "install": [[sys.executable, "-m", "pip", "install", "yamllint"]],
+            "help": ["yamllint", "-h"],
+            "run": ["yamllint", "-f", "parsable", "."],
+            "rundefault": [
+                "yamllint",
+                "-c",
+                "{config_dir}/.yamllint",
+                "-f",
+                "parsable",
+                ".",
+            ],
+            "dotfiles": [".yamllint"],
+            "parser": parsers.YAMLLintParser,
+            "language": "yaml",
+            "autorun": True,
+            "run_per_file": False,
+        },
+    )
+
+
+def init():
```

Comment:
I'll be moving a lot of the functionality in here into a class, these could go into the init.